keyboards/matrix/pjf/rules.mk

```diff
@@ -45,5 +45,9 @@ CONSOLE_ENABLE = no# Console for debug
 COMMAND_ENABLE = no # Commands for debug and configuration
 NKRO_ENABLE = no # USB Nkey Rollover
 AUDIO_ENABLE = no
-
 RGBLIGHT_ENABLE = yes
+
+RGBLIGHT_CUSTOM_DRIVER = yes
+CUSTOM_MATRIX = yes
+
+SRC += matrix.c drivers/arm/ws2812.c
```

Comment:
Directly including `arm/ws2812.c` and using `RGBLIGHT_CUSTOM_DRIVER` for some board specific logic seems rather unclean. Given the same thing looks to be achievable with `rgblight_set_clipping_range` and `led_update_kb`.

Reply:
Seems there lacks of hooks for customize the rgblight before sent to ws2812.  led_update_kb() was an event trigger function, but rgblight_task() will be called periodically, thus made it's hard to sync the light state.